tfhe/src/shortint/key_switching_key/mod.rs

```diff
@@ -122,7 +125,10 @@ impl KeySwitchingKey {
 
             // Cast to smaller bit length: left shift, then keyswitch
             i if i < 0 => {
-                let acc = self.src_server_key.generate_lookup_table(|n| n << -i);
+                // We want to avoid the padding bit to be dirty, hence the modulus
+                let acc = self.src_server_key.generate_lookup_table(|n| {
+                    (n << -i) % (ct.carry_modulus.0 * ct.message_modulus.0) as u64
```

Comment:
does this make sense functionally ? I'm not sure about what the casting was doing, if we are trashing the padding bit we may want to check what's going on